socorro/external/crashstorage_base.py

```diff
@@ -195,140 +188,98 @@ def is_mutator(self):
 
-        parameters:
-           crash_id - the id of a raw crash to fetch"""
+        :param crash_id: crash report id for data to fetch
+
+        :returns: DotDict of raw crash data
+
+        """
         raise NotImplementedError("get_raw_crash is not implemented")
 
     def get_raw_dump(self, crash_id, name=None):
-        """the default implementation of fetching a dump
+        """Fetch dump for a raw crash
 
-        parameters:
-           crash_id - the id of a dump to fetch
-           name - the name of the dump to fetch"""
+        :param crash_id: crash report id
+        :param name: name of dump to fetch
+
+        :returns: dump as bytes
+
+        """
         raise NotImplementedError("get_raw_dump is not implemented")
 
     def get_raw_dumps(self, crash_id):
-        """the default implementation of fetching all the dumps
+        """Fetch all dumps for a crash report
 
-        parameters:
-           crash_id - the id of a dump to fetch"""
+        :param crash_id: crash report id
+
+        :returns: MemoryDumpsMapping of dumps
+
+        """
         raise NotImplementedError("get_raw_dumps is not implemented")
 
     def get_raw_dumps_as_files(self, crash_id):
-        """the default implementation of fetching all the dumps as files on
-        a file system somewhere.  returns a list of pathnames.
+        """Fetch all dumps for a crash report and save as files.
 
-        parameters:
-           crash_id - the id of a dump to fetch"""
+        :param crash_id: crash report id
+
+        :returns: dict of dumpname -> file path
+
+        """
         raise NotImplementedError("get_raw_dumps is not implemented")
 
     def get_processed(self, crash_id):
-        """the default implementation of fetching a processed_crash.  This
-        method should not be overridden in subclasses unless the intent is to
-        alter the redaction process.
+        """Fetch processed crash.
+
+        :arg crash_id: crash report id
 
-        :arg crash_id: the id of a processed_crash to fetch
+        :returns: DotDict
 
         """
         processed_crash = self.get_unredacted_processed(crash_id)
         self.redactor(processed_crash)
         return processed_crash
 
     def get_unredacted_processed(self, crash_id):
-        """the implementation of fetching a processed_crash with no redaction
+        """Fetch processed crash with no redaction
 
-        parameters:
-           crash_id - the id of a processed_crash to fetch"""
+        :param crash_id: crash report id
+
+        :returns: DotDict
+
+        """
         raise NotImplementedError("get_unredacted_processed is not implemented")
 
     def remove(self, crash_id):
-        """delete a crash from storage
+        """Delete crash report data from storage
 
-        parameters:
-           crash_id - the id of a crash to fetch"""
-        raise NotImplementedError("remove is not implemented")
+        :param crash_id: crash report id
 
-    def new_crashes(self):
```

Comment:
This and `ack_crash` were from the days when crashstorage was also used for queueing crashes for processing. I split that out into a separate crashqueue hierarchy, so this is vestigial now.